x-pack/plugins/elastic_assistant/server/lib/langchain/execute_custom_llm_chain/index.test.ts

```diff
@@ -209,5 +228,55 @@ describe('callAgentExecutor', () => {
         false
       );
     });
+
+    it('does not streams token after handleStreamEnd has been called', async () => {
+      const mockInvokeWithChainCallback = jest.fn().mockImplementation((a, b, c, d, e, f, g) => {
+        b.callbacks[0].handleLLMNewToken('hi', {}, '123', '456');
+        b.callbacks[0].handleChainEnd({ output: 'hello' }, '123');
+        b.callbacks[0].handleLLMNewToken('hey', {}, '678', '456');
+        return Promise.resolve();
+      });
+      (initializeAgentExecutorWithOptions as jest.Mock).mockImplementation(
+        (_a, _b, { agentType }) => ({
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          call: (props: any, more: any) => mockCall({ ...props, agentType }, more),
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          invoke: (props: any, more: any) =>
+            mockInvokeWithChainCallback({ ...props, agentType }, more),
+        })
+      );
+      const onLlmResponse = jest.fn();
+      await callAgentExecutor({ ...defaultProps, onLlmResponse, isStream: true });
+
+      expect(mockPush).toHaveBeenCalledWith({ payload: 'hi', type: 'content' });
+      expect(mockPush).not.toHaveBeenCalledWith({ payload: 'hey', type: 'content' });
+    });
+
+    it('only streams tokens with length from the root parentRunId', async () => {
```

Comment:
this test confirms the fix. The test above it I added because we didn't have it, and not related to the bug.